### PR TITLE
(build) fix JDK 16/17 builds

### PR DESCRIPTION
**Pull request by bmarwell**
* Replaced a few EasyMock tests with Mockito where needed.
* `--illegal-access=permit` for guice tests.
* Remove JDK17-ea for now, no groovy support.

---

Following this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SHIRO) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SHIRO-XXX] - Fixes bug in SessionManager`,
       where you replace `SHIRO-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 
Trivial changes like typos do not require a JIRA issue (javadoc, comments...). 
In this case, just format the pull request title like `(DOC) - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bmarwell** [comment]
The PR will only fail for JDK 11, but will work on JDK 8 and 14:

* https://ci-builds.apache.org/job/Shiro/job/Shiro-jdk8/view/change-requests/job/PR-278/
* https://ci-builds.apache.org/job/Shiro/job/Shiro-jdk11/view/change-requests/job/PR-278/ (that is this toggling test… works just sometimes.)
* https://ci-builds.apache.org/job/Shiro/job/Shiro-jdk14/view/change-requests/job/PR-278/

File: integration-tests/guice3/pom.xml [hunk 1]
```diff
@@ -123,4 +123,16 @@
...
 	</dependencies>
 
+	<profiles>
+		<profile>
+			<id>jdk16</id>
+			<activation>
+				<jdk>[16,)</jdk>
+			</activation>
+			<properties>
+				<surefire.argLine>--illegal-access=permit</surefire.argLine>
```
Comment (bdemers) [inline integration-tests/guice3/pom.xml:133]: nit: add a comment as to why this is needed (to give us a hint when to remove it in the future)

File: pom.xml [hunk 2]
```diff
@@ -70,6 +70,10 @@
         <!-- Replaced by the build number plugin at build time: -->
         <buildNumber>${user.name}-${maven.build.timestamp}</buildNumber>
         <jacoco.skip>true</jacoco.skip>
+        <!--suppress CheckTagEmptyBody -->
+        <surefire.argLine></surefire.argLine>
```
Comment (bdemers) [inline pom.xml:74]: I'm guessing this works, but (if you haven't already) double-check to make sure the code coverage works (Jacoco sets the `argLine` prop too)

File: pom.xml [hunk 3]
```diff
@@ -1400,6 +1414,12 @@
                 </pluginManagement>
             </build>
         </profile>
+        <profile>
+            <id>jdk16</id>
+            <activation>
+                <jdk>[16,)</jdk>
+            </activation>
+        </profile>
```
Comment (bdemers) [inline pom.xml:1422]: delete the empty profile block?

**Review by bdemers — APPROVED** [review]
a couple of minor nits, but looks good to me!

idea: we may want to make a note in the .github/template to look for conversions from easy-mock to mockito

(I'm not sure that's the _best_ place for it) but it does bring focus to the effort (and adds a doc point as to why we are doing it). NOTE: that doesn't need to block this from getting merged though

**Review by fpapon — APPROVED** [review]
LGTM

**Pull request merged**